contracts/dnsregistrar/OffchainDNSResolver.sol

```diff
@@ -199,10 +200,24 @@ contract OffchainDNSResolver is IExtendedResolver, IERC165 {
         uint256 startIdx,
         uint256 lastIdx
     ) internal pure returns (bytes memory) {
-        // TODO: Concatenate multiple text fields
-        uint256 fieldLength = data.readUint8(startIdx);
-        assert(startIdx + fieldLength < lastIdx);
-        return data.substring(startIdx + 1, fieldLength);
+        uint256 totalLength = 0;
+        uint256 idx = startIdx;
+        while (idx < lastIdx) {
+            uint256 fieldLength = data.readUint8(idx);
+            totalLength += fieldLength;
+            idx += fieldLength + 1;
+        }
+
+        bytes memory result = new bytes(totalLength);
+        idx = startIdx;
+        uint256 resultIdx = 0;
+        while (idx < lastIdx) {
+            uint256 fieldLength = data.readUint8(idx);
+            result.strcpy(resultIdx, data, idx + 1, fieldLength);
+            resultIdx += fieldLength;
+            idx += fieldLength + 1;
+        }
```

Comment:
```suggestion
        uint256 resultIdx = 0;
        for (uint256 idx = startIdx; idx < lastIdx; idx += fieldLength + 1) {
            uint256 fieldLength = data.readUint8(idx);
            result.strcpy(resultIdx, data, idx + 1, fieldLength);
            resultIdx += fieldLength;
        }
```
```

Reply:
made the same change here as well;
https://github.com/ensdomains/ens-contracts/pull/332#discussion_r1844368380